simplic-authentication/Simplic.Authentication.Service/AuthenticationService.cs

```diff
@@ -41,45 +45,133 @@ public AuthenticationService(IUnityContainer container, IAuthorizationService au
+        /// </summary>
+        /// <param name="domain">Current domain</param>
+        /// <param name="userName">Current user</param>
+        /// <param name="password">Current password</param>
+        public void SetAutologin(string domain, string userName, string password)
+        {
+            try
+            {
+                var path = GetAutologinPath();
+
+                if (Login(domain, userName, password) != null)
+                {
+                    // Create autologin object
+                    var hash = GenerateHash(userName, domain);
+                    var loginObject = new AutologinModel() { Hash = hash, UserName = userName, Domain = domain };
+
+                    // Ensure directory
+                    IO.DirectoryHelper.CreateDirectoryIfNotExists(Path.GetDirectoryName(GetAutologinPath()));
+
+                    File.WriteAllText(GetAutologinPath(), JsonConvert.SerializeObject(loginObject));
+                }
+            }
+            catch
+            {
+                /* swallow */
+            }
+        }
+
+        /// <summary>
+        /// Remove autologin for the current windows user
+        /// </summary>
+        public void RemoveAutologin()
+        {
+            var path = GetAutologinPath();
+            if (File.Exists(path))
+                File.Delete(path);
+        }
+
+        /// <summary>
+        /// Check whether autologin is existing and valid for the current user
+        /// </summary>
+        /// <returns>Simplic session if login was successfull else or null</returns>
+        public Session TryAutologin()
+        {
+            var path = GetAutologinPath();
+            if (File.Exists(path))
+            {
+                var obj = JsonConvert.DeserializeObject<AutologinModel>(File.ReadAllText(path));
+                if (GenerateHash(obj.UserName, obj.Domain) == obj.Hash)
+                {
+                    var user = userService.GetByName(obj.UserName);
+                    return GenerateUserSession(user);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Generate unique hash
+        /// </summary>
+        /// <returns>Unique hash for the given user</returns>
+        private string GenerateHash(string userName, string domain)
+        {
+            return Security.Cryptography.CryptographyHelper.HashSHA256(NetworkInterface
+                   .GetAllNetworkInterfaces()
+                   .Where(nic => nic.OperationalStatus == OperationalStatus.Up && nic.NetworkInterfaceType != NetworkInterfaceType.Loopback)
+                   .Select(nic => nic.GetPhysicalAddress().ToString())
+                   .FirstOrDefault() + $"{Environment.UserDomainName}_{Environment.UserName}_{userName}_{domain}");
+        }
+
+        /// <summary>
+        /// Get autologin file path
+        /// </summary>
+        /// <returns>Autologin file path</returns>
+        private string GetAutologinPath()

```

Comment:
why is this a method ? it could easily be a private field or better yet readonly variable.